src/cudamatrix/cu-device.h

```diff
@@ -55,7 +55,12 @@ class CuDevice {
   inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
 
   inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
-    return allocator_.MallocPitch(row_bytes, num_rows, pitch);
+    if (random_stride_mode_) {
+      return allocator_.MallocPitch(row_bytes + 128 * (rand() & 1), num_rows,
```

Comment:
I see.  For testing purposes any number >0 is OK here though, because the
matrix sizes are randomized, so for some matrix sizes, 128 is sure to put
them in different buckets.  So it's fine.

On Wed, Aug 24, 2016 at 1:38 AM, Shiyin Kang <notifications@github.com>
wrote:

> In src/cudamatrix/cu-device.h
> <https://github.com/kaldi-asr/kaldi/pull/997#discussion_r75997292>:
>
> > @@ -55,7 +55,12 @@ class CuDevice {
> >    inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
> >
> >    inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
> > -    return allocator_.MallocPitch(row_bytes, num_rows, pitch);
> > +    if (random_stride_mode_) {
> > +      return allocator_.MallocPitch(row_bytes + 128 * (rand() & 1), num_rows,
>
> The stride bucket size is hardware dependent. It seems it is keep
> increasing. On K40c with CUDA 7.5 it is 512.
> http://docs.nvidia.com/cuda/cuda-c-programming-guide/
> index.html#device-memory-accesses
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/997/files/ccd2b85ea256cfe1e3f6ad76c04c4e260a0e6344#r75997292>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu4eXLKo3QelL8ET_esyZNlEhkisZks5qi9jygaJpZM4Jqu95>
> .
>